test/model/content_test.dart

```diff
@@ -420,12 +421,52 @@ class ContentExample {
 
   static const imageSingle = ContentExample(
     'single image',
+    // https://chat.zulip.org/#narrow/stream/7-test-here/topic/Thumbnails/near/1893590
+    "[image.jpg](/user_uploads/2/c3/wb9FXk8Ej6qIc28aWKcqUogD/image.jpg)",
+    '<div class="message_inline_image">'
+      '<a href="/user_uploads/2/c3/wb9FXk8Ej6qIc28aWKcqUogD/image.jpg" title="image.jpg">'
+        '<img src="/user_uploads/thumbnail/2/c3/wb9FXk8Ej6qIc28aWKcqUogD/image.jpg/840x560.webp"/></a></div>', [
+    ImageNodeList([
+      ImageNode(srcUrl: '/user_uploads/2/c3/wb9FXk8Ej6qIc28aWKcqUogD/image.jpg',
+        thumbnailUrl: '/user_uploads/thumbnail/2/c3/wb9FXk8Ej6qIc28aWKcqUogD/image.jpg/840x560.webp',
+        loading: false),
+    ]),
+  ]);
+
+  static const imageSingleNoThumbnail = ContentExample(
+    'single image no thumbnail',
     "https://chat.zulip.org/user_avatars/2/realm/icon.png?version=3",
     '<div class="message_inline_image">'
       '<a href="https://chat.zulip.org/user_avatars/2/realm/icon.png?version=3">'
         '<img src="https://chat.zulip.org/user_avatars/2/realm/icon.png?version=3"></a></div>', [
     ImageNodeList([
-      ImageNode(srcUrl: 'https://chat.zulip.org/user_avatars/2/realm/icon.png?version=3'),
+      ImageNode(srcUrl: 'https://chat.zulip.org/user_avatars/2/realm/icon.png?version=3',
+        thumbnailUrl: null, loading: false),
+    ]),
+  ]);
+
+  static const imageSingleLoadingPlaceholder = ContentExample(
+    'single image loading placeholder',
+    // https://chat.zulip.org/#narrow/stream/7-test-here/topic/Thumbnails/near/1893590
+    "[image.jpg](/user_uploads/2/c3/wb9FXk8Ej6qIc28aWKcqUogD/image.jpg)",
+    '<div class="message_inline_image">'
+      '<a href="/user_uploads/2/c3/wb9FXk8Ej6qIc28aWKcqUogD/image.jpg" title="image.jpg">'
+        '<img class="image-loading-placeholder" src="/static/images/loading/loader-black.svg"></a></div>', [
+    ImageNodeList([
+      ImageNode(srcUrl: '/user_uploads/2/c3/wb9FXk8Ej6qIc28aWKcqUogD/image.jpg',
+        thumbnailUrl: null, loading: true),
+    ]),
+  ]);
+
+  static const imageSingleExternal = ContentExample(
+    'single image external',
+    "https://upload.wikimedia.org/wikipedia/commons/7/78/Verregende_bloem_van_een_Helenium_%27El_Dorado%27._22-07-2023._%28d.j.b%29.jpg",
```

Comment:
Is there an example message for this on chat.zulip.org which you can link to?

Reply:
Ha, yeah I'd just noticed I forgot to add that and pushed a new revision with that.